cmd/cmd_controller.js

```diff
@@ -263,25 +273,58 @@ class EmbarkController {
           engine.logger.info(__("loaded plugins") + ": " + pluginList.join(", "));
         }
 
-        engine.startService("processManager");
-        engine.startService("serviceMonitor");
-        engine.startService("libraryManager");
-        engine.startService("codeRunner");
-        engine.startService("web3");
-        engine.startService("pipeline");
-        engine.startService("deployment", {onlyCompile: false});
-        engine.startService("storage");
-        engine.startService("codeGenerator");
-        engine.startService("fileWatcher");
-
-        callback();
+        engine.ipc.connect((err) => {
+          if (err) {
+            engine.startService("processManager");
+            engine.startService("serviceMonitor");
+            engine.startService("libraryManager");
+            engine.startService("codeRunner");
+            engine.startService("web3");
+            engine.startService("pipeline");
+            engine.startService("deployment");
+            engine.startService("storage");
+            engine.startService("codeGenerator");
+            engine.startService("webServer");
+
+            return callback();
+          }
+
+          engine.startService("codeRunner");
+          callback();
+        });
+      },
+      function web3IPC(callback) {
+        if(!engine.ipc.connected || engine.ipc.isServer()) {
```

Comment:
The function web3IPC is here in case the console is using IPC.
In that case, there is no need for deploying contracts....

But we need to setup the context and web3.
Why web3: because it cannot be passed via message (IPC).
The history, to get access to all contracts